### NLP project

In [2]:
import pandas as pd
import os

In [20]:
data = pd.read_json("Arts_Crafts_and_Sewing_5.json", lines= True)

In [6]:
data.head(5)

overall  verified   reviewTime      reviewerID        asin  \
0        4      True  03 29, 2016   AIE8N9U317ZBM  0449819906   
1        5      True  08 12, 2015  A3ECOW0TWLH9V6  0449819906   
2        4      True   04 5, 2015  A278N8QX9TY2OS  0449819906   
3        5      True  10 11, 2014  A123W8HIK76XCN  0449819906   
4        5      True   05 8, 2014  A2A6MZ2QB4AE0L  0449819906   

                            style            reviewerName  \
0  {'Format:': ' Kindle Edition'}          Zelmira, Ph.D.   
1       {'Format:': ' Paperback'}  Dangerous when Cooking   
2       {'Format:': ' Paperback'}                 Just us   
3  {'Format:': ' Kindle Edition'}         Amazon Customer   
4       {'Format:': ' Paperback'}          Sustainability   

                                          reviewText  \
0                Contains some interesting stitches.   
1  I'm a fairly experienced knitter of the one-co...   
2  Great book but the index is terrible. Had to w...   
3  I purchased the Kindle edition which is incred...   
4  Very well laid out and very easy to read.\n\nT...   

                                             summary  unixReviewTime vote  \
0                                         Four Stars      1459209600  NaN   
1    My current favorite go-to guide for inspiration      1439337600   18   
2  lots of great examples, good instructions, col...      1428192000    3   
3              Another little gem by Melissa Leapman      1412985600  NaN   
4                                 Very comprehensive      1399507200  NaN   

  image  
0   NaN  
1   NaN  
2   NaN  
3   NaN  
4   NaN

In [13]:
print(data.shape)
print(data.columns)
print(data.dtypes)

(494485, 12)
Index(['overall', 'verified', 'reviewTime', 'reviewerID', 'asin', 'style',
       'reviewerName', 'reviewText', 'summary', 'unixReviewTime', 'vote',
       'image'],
      dtype='object')
overall            int64
verified            bool
reviewTime        object
reviewerID        object
asin              object
style             object
reviewerName      object
reviewText        object
summary           object
unixReviewTime     int64
vote              object
image             object
dtype: object


In [25]:
data.image[26]

['https://images-na.ssl-images-amazon.com/images/I/51zwvBnXUUL._SY88.jpg',
 'https://images-na.ssl-images-amazon.com/images/I/51-LYgXa3IL._SY88.jpg',
 'https://images-na.ssl-images-amazon.com/images/I/61P4n-DuUHL._SY88.jpg']

In [18]:
#data['vote'] = data['vote'].astype('float64')

type(data.vote[2])

str

In [12]:
print(data.verified.value_counts(normalize = True))

True     0.938864
False    0.061136
Name: verified, dtype: float64


In [7]:
print('reviewer names : ', data.reviewerName.nunique())
print('reviewerID : ', data.reviewerID.nunique())

reviewer names :  45478
reviewerID :  56210


There are only 45478 reviewers amongst the 494485 reviews : it's always the same people who comment on Amazon. Or rather, once you put a review on Amazon, it's more likely you'll do it again. 

In [10]:
# Some useless variables ?  no just wrong format

print('image missing value : ', len(data.image.isnull()))
print('vote missing value : ', len(data.vote.isnull()))

image missing value :  494485
vote missing value :  494485


In [11]:
data.describe()

overall  unixReviewTime
count  494485.000000    4.944850e+05
mean        4.566852    1.450450e+09
std         0.929987    4.548772e+07
min         1.000000    1.084666e+09
25%         5.000000    1.422058e+09
50%         5.000000    1.454198e+09
75%         5.000000    1.483920e+09
max         5.000000    1.538525e+09

### Are reviews on Amazon rather positive, negative, neutral ? Is there a trend to spot ?

Two main variables will be analysed for our sentiment analysis problem: `reviewText` and `summary`.